internal/resources/cluster/integration/resource_integration.go

```diff
@@ -0,0 +1,226 @@
+	if err != nil {
+		if clienterrors.IsNotFoundError(err) {
+			d.SetId("")
+			return
+		}
+
+		return diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control integration entry, name: %s", fn.Name))
+	}
+
+	d.SetId(resp.Integration.Meta.UID)
+
+	if err = d.Set(common.MetaKey, common.FlattenMeta(resp.Integration.Meta)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diags
+}
+
+func ResourceIntegration() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIntegrationCreate,
+		ReadContext:   dataSourceIntegrationRead,
+		UpdateContext: resourceIntegrationInPlaceUpdate,
+		DeleteContext: resourceIntegrationDelete,
+		Schema:        integrationSchema,
+	}
+}
+
+func resourceIntegrationCreate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	return nil
+}
+
+func resourceIntegrationInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	return nil
+}
+
+func resourceIntegrationDelete(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	return nil
+}
+
+var integrationSchema = map[string]*schema.Schema{
+	managementClusterNameKey: {
+		Type:        schema.TypeString,
+		Description: "Name of the management cluster",
+		Required:    true,
+		ForceNew:    true,
+	},
+	provisionerNameKey: {
+		Type:        schema.TypeString,
+		Description: "Provisioner of the cluster",
+		Required:    true,
+		ForceNew:    true,
+	},
+	clusterNameKey: {
+		Type:        schema.TypeString,
+		Description: "Name of the cluster",
+		Required:    true,
+		ForceNew:    true,
+	},
+	nameKey: {
+		Type:        schema.TypeString,
+		Description: "Name of the integration",
+		Required:    true,
+		ForceNew:    true,
+	},
+
+	common.MetaKey: common.Meta,
+
+	specKey: integrationSpec,
+
+	statusKey: {
+		Type:        schema.TypeMap,
+		Description: "Status of integraion resource",
+		Computed:    true,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+	},
+}
+
+var integrationSpec = &schema.Schema{
+	Type:        schema.TypeMap,
+	Description: "Integration Spec",
+	Required:    true,
+	ForceNew:    true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			configurationsKey: {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						enableNamespaceExclusionsKey: {
+							Type:        schema.TypeBool,
+							Description: "Enable Namespace Exclusions",
+							Optional:    true,
+							ForceNew:    true,
+						},
+					},
+				},
+			},
+		},
+	},
+}
+
+func constructFullName(d *schema.ResourceData) *integration.VmwareTanzuManageV1alpha1ClusterIntegrationFullName {
+	fullname := &integration.VmwareTanzuManageV1alpha1ClusterIntegrationFullName{}
+
+	if value, ok := d.GetOk(managementClusterNameKey); ok {
+		fullname.ManagementClusterName = value.(string)
```

Comment:
to avoid runtime panics due to type conversions, please use 
fullname.ManagementClusterName , _ = value.(string)